libs/api/domains/national-registry/src/lib/v1/soffia.service.ts

```diff
@@ -0,0 +1,416 @@
+            nationalId: familyMember.Kennitala,
+            gender: mapGender(familyMember.Kyn),
+          } as FamilyMember),
+      )
+      .sort((a, b) => {
+        return (
+          kennitala.info(b.nationalId).age - kennitala.info(a.nationalId).age
+        )
+      })
+
+    return members
+  }
+
+  async getFamilyMemberDetails(
+    nationalId: User['nationalId'],
+    familyMemberNationalId: User['nationalId'],
+  ): Promise<FamilyChild> {
+    const family = await this.nationalRegistryApi.getMyFamily(nationalId)
+    const isAllowed = some(family, ['Kennitala', familyMemberNationalId])
+    /**
+     * Only show data if SSN is part of family.
+     */
+    if (isAllowed) {
+      const familyMember = await this.nationalRegistryApi.getUser(
+        familyMemberNationalId,
+      )
+      return {
+        fullName: familyMember.Fulltnafn,
+        firstName: familyMember.Eiginnafn,
+        nationalId: familyMemberNationalId,
+        gender: familyMember.Kyn,
+        displayName: familyMember.Birtnafn,
+        middleName: familyMember.Millinafn,
+        surname: familyMember.Kenninafn,
+        lastName: familyMember.Kenninafn,
+        genderDisplay: familyMember.Kynheiti,
+        birthday: familyMember.Faedingardagur,
+        parent1: familyMember.Foreldri1,
+        nameParent1: familyMember.nafn1,
+        parent2: familyMember.Foreldri2,
+        nameParent2: familyMember.Nafn2,
+        custody1: undefined,
+        nameCustody1: undefined,
+        custodyText1: undefined,
+        custody2: undefined,
+        nameCustody2: undefined,
+        custodyText2: undefined,
+        birthplace: familyMember.Faedingarstadur,
+        religion: familyMember.Trufelag,
+        nationality: familyMember.RikisfangLand,
+        homeAddress: familyMember.Logheimili,
+        municipality: familyMember.LogheimiliSveitarfelag,
+        postal: `${familyMember.Postnr} ${familyMember.LogheimiliSveitarfelag}`, // Same structure as familyChild.Postaritun
+        fate: familyMember.Afdrif1 || familyMember.Afdrif2,
+      }
+    } else {
+      throw new ForbiddenException('Family member not found')
+    }
+  }
+
+  async getChildren(
+    nationalId: User['nationalId'],
+    data?: V1RawData,
+  ): Promise<FamilyChild[]> {
+    const myChildren =
+      data?.children ??
+      (await this.nationalRegistryApi.getMyChildren(nationalId))
+
+    const members = myChildren
+      .filter((familyChild) => {
+        const isNotUser = familyChild.Barn !== nationalId
+        const isUnderEighteen = kennitala.info(familyChild.Barn).age < 18
+
+        return isNotUser && isUnderEighteen
+      })
+      .map((familyChild) => formatFamilyChild(familyChild))
+      .filter(isDefined)
+      .sort((a, b) => {
+        return (
+          kennitala.info(b.nationalId).age - kennitala.info(a.nationalId).age
+        )
+      })
+
+    return members
+  }
+
+  async getChildCustody(
+    nationalId?: string,
```

Comment:
Why is `nationalId` allowed optiona?
Both in this method and `getParents`

Reply:
If `data` is provided the nationalId is optional since i get the child data from there